mbuild/port.py

```diff
@@ -94,6 +135,11 @@ def direction(self):
         """
         return unit_vector(self.xyz_with_ports[1]-self.xyz_with_ports[0])
 
+    @property
+    def separation(self):
+        if self.anchor:
```

Comment:
I figure it should return None if there isn't an anchor particle since this function specifically gives the distance between a port and its anchor particle. I included a warning message as well in the latest commit.